Library/Homebrew/rubocops/on_os_block.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+require "rubocops/order_utils"
+require "rubocops/extend/formula"
+
+module RuboCop
+  module Cop
+    module FormulaAudit
+      class OnOsBlock < FormulaCop
```

Comment:
This could live in the same file (if not same class) as `ComponentsOrder`. I strongly suspect you've broken those tests in extracting to separate methods. It may be worth making just the changes to `ComponentsOrder` first and then extracting the other stuff in a follow-up PR. It'll make it easier to review and to test.

Reply:
Found it. I better understand that part now. Mainly, the `component_problem` call can be moved out of my new check_order function (separation of concerns). I'll update the PR.